dali/base/dafdesc.cpp

```diff
@@ -1353,14 +1357,29 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
                 *_interleaved = interleaved;
         }
 
-        const char * historyStr = "History";
-        if (mb.remaining() > strlen(historyStr))
+        // If we have more serialised data in MemoryBuffer
+        if (mb.remaining() >= sizeof(unsigned short))
         {
-            const char * buffer = mb.toByteArray()+mb.getPos();
-            if (nullptr != strstr(buffer, historyStr))
-                history.setown(createPTree(mb));
+            unsigned short extendedVersion;
+            mb.read(extendedVersion);
+
+            if (extendedVersion >= VERSION_MINHISTORY)
+            {
+                //If extendedVersion >= 1 then we should have stored History
+                const char * historyStr = "History";
+                if (mb.remaining() > strlen(historyStr))
+                {
+                    const char * buffer = mb.toByteArray()+mb.getPos();
+                    if (nullptr != strstr(buffer, historyStr))
+                        history.setown(createPTree(mb));
+                    else
+                        history.setown(createPTree(historyStr));
```

Comment:
>Your suggestion is wrong. That code fragment doesn't deserialise the history, but create a new and empty one.

That was a typo'. I should of course have read:
```

You shouldn't be manually checking (strstr etc), just deserialize it. You have already validated the version. So:

if (extendedVersion >= VERSION_MINHISTORY)
    history.setown(createPTree(mb));
else
    throw ...
```

I can see no good reason to do further [partial] validation of the content of mb, with a strlen and a strstr call, beyond the version check. No more so than for any other part of deserializing the file descriptor or for that matter in any other serialize/deserialize code elsewhere in the system.

Sure it could be corrupted, truncated, etc. etc. - but if there were issues like that, then it could hit deserialize problems at any stage of the the deserializtion of the file descriptor, History isn't special in that sense.
It doesn't make sense to add this kind of manual partial checking of the content of serialized data ahead of deserializing it everywhere.

Now, you _could_ argue that validation and error handling *in* the deserialize methods should be more robust, but that's a separate and general issue.

After successful deserialization you _could_ add a assertex(streq("History", history->queryName());

The same applies to try/catch, you could, but unexpected result could happen during any part of the deserialization, not just History and it would be impossible to recover satisfactorily in most cases, nor would it be desirable in most cases to continue.

@AttilaVamos - if you have some concerns as to why History is more likely to be corrupted during serialization/deserialization than any other part of the serialization/deserialization, then.. maybe I've missed something. And if you really think it's important that these partial memory buffer checks remain ahead of deserializing it for History specifically then leave them in and add a comment in the code as to why these additional checks are done here for History.